impl/local/localindexrepo.go

```diff
@@ -0,0 +1,111 @@
+package local
```

Comment:
Let's avoid impl packages; not idiomatic in Go. These should go in the `index` package directly. It's OK for it to contain the interface _and_ the implementations.